### ISSUE TITLE
outputBy process does not take into [Namespace] attribute

### ISSUE THREAD

**Issue by geoffreymcgill** [bridgedotnet/Bridge#104]
If the `[Namespace]` attribute is set, the `outputBy` processing still uses the C# namespace name. 

Give the following sample, expected filename would be `utils.js`, but actual output is `demo.utilities.js`. 

#### Demo.cs

``` csharp
using Bridge;

namespace Demo.Utilities
{
    [Namespace("Utils")]
    public class App
    {
        public string Test { get; set; }
    }
}
```

**Comment by fabriciomurta** [comment]
This seems to be a defect inherited from `Mono.Cecil`.

When we do:
`assembly=Mono.Cecil.ReadReference()`, (Translator.InspectAssembly.cs:15)

the `assembly.MainModule.Type` returned has the namespace attribute as 'Demo.Utilities', 

`assembly.MainModule.Type.CustomAttributes` in turn has an `NamespaceAttribute`

This custom attribute `NamespaceAttribute` has the `Utils` argument specified as expected.

**Counter-question is:** _can this be fixed on bridge level?_
That would be: replace wrong namespace on type by specified namespace attribute whenever the custom attribute is present

Or is this a fix we should do on Mono.Cecil? I'm not sure if the affected code is in Mono.Cecil (closed-source-dll?) or NRefactory, which we can change and fix.

**Comment by fabriciomurta** [comment]
By vladimir's orientation, the Mono.Cecil is behaving as expected -- attributes are bridge-specific and should not be dealt by from Cecil.

To have access to the custom attributes where the file name is defined, on EmitBlock.cs (`Bridge.Translator.EmitBlock.GetOutputForType()`), this can be used:
`var cas = this.Emitter.GetTypeDefinition(typeInfo.TypeDeclaration.ReturnType).CustomAttributes;`

So rest of the logic is straight forward.

**Comment by DaniilVeriga** [comment]
@fabriciomurta, do you know there is any milestone for this issue?

**Comment by fabriciomurta** [comment]
@DaniilVeriga I'm not sure... might it be 1.3 as it is going to be released on the upcoming 1.3 version?

**Comment by DaniilVeriga** [comment]
@fabriciomurta, yes, if it is going to be released with 1.3, then a milestone should be 1.3. I just didn't know what release it is going to included into:)
